### PR TITLE
[JBPM-9592] External DB driver not working-MySQL

### PR DESCRIPTION
**Pull request by fjtirado**
Code now does not longer assume that a Date instance is received, but
also contemplates LocalDateTime, Number and String cases for
completeness

### CODE REVIEW

File: kie-soup-dataset/kie-soup-dataset-sql/src/main/java/org/dashbuilder/dataprovider/sql/dialect/DefaultDialect.java [hunk 1]
```diff
@@ -114,13 +118,23 @@ public Double convertToDouble(Object value) {
...
+			return (Date) value;
+		} else if (value instanceof LocalDateTime) {
+			return Timestamp.valueOf((LocalDateTime) value);
+		} else if (value instanceof Number) {
+			return new Date(((Number) value).longValue());
+		} else {
+			try {
+				return DateFormat.getDateTimeInstance().parse(value.toString());
+			} catch (ParseException e) {
+				throw new IllegalArgumentException("Unable to convert " + value + "of type "+value.getClass()+" to date", e);
```
Comment (afalhambra) [inline kie-soup-dataset/kie-soup-dataset-sql/src/main/java/org/dashbuilder/dataprovider/sql/dialect/DefaultDialect.java:134]:
```suggestion
				throw new IllegalArgumentException("Unable to convert " + value + " of type "+value.getClass()+" to date", e);
```

**Review by afalhambra — APPROVED** [review]
LGTM

**Comment by danielezonca** [comment]
Is it possible to add a test?

**Comment by fjtirado** [comment]
> Is it possible to add a test?

We will need the particula mysql instance that now returns something different than Date, so I would say no, the change is pretty straighforward

**Pull request merged**